app/scripts/lib/multichain-api/adapters/caip-permission-adapter-permittedChains.test.ts

```diff
@@ -0,0 +1,314 @@
+        '0x65',
+      );
+
+      expect(result).toStrictEqual({
+        requiredScopes: {
+          'eip155:1': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+          },
+        },
+        optionalScopes: {
+          'eip155:100': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:100:0x100'],
+          },
+          'eip155:101': {
+            methods: validRpcMethods,
+            notifications: validNotifications,
+            accounts: [],
+          },
+        },
+        isMultichainOrigin: false,
+      });
+    });
+
+    it('does not modify the input CAIP-25 caveat value object', () => {
+      const input: Caip25CaveatValue = {
+        requiredScopes: {
+          'eip155:1': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+          },
+        },
+        optionalScopes: {},
+        isMultichainOrigin: false,
+      };
+
+      const result = addPermittedEthChainId(input, '0x65');
+
+      expect(input).toStrictEqual({
+        requiredScopes: {
+          'eip155:1': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+          },
+        },
+        optionalScopes: {},
+        isMultichainOrigin: false,
+      });
+      expect(input).not.toStrictEqual(result);
+    });
+
+    it('does not add an optional scope for the chainId if already exists in the required scopes', () => {
+      const input: Caip25CaveatValue = {
+        requiredScopes: {
+          'eip155:1': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+          },
+        },
+        optionalScopes: {
+          'eip155:100': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:100:0x100'],
+          },
+        },
+        isMultichainOrigin: false,
+      };
+      const result = addPermittedEthChainId(input, '0x1');
+
+      expect(result).toStrictEqual(input);
+    });
+
+    it('does not add an optional scope for the chainId if already exists in the optional scopes', () => {
+      const input: Caip25CaveatValue = {
+        requiredScopes: {
+          'eip155:1': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+          },
+        },
+        optionalScopes: {
+          'eip155:100': {
+            methods: [],
+            notifications: [],
+            accounts: ['eip155:100:0x100'],
+          },
+        },
+        isMultichainOrigin: false,
+      };
+      const result = addPermittedEthChainId(input, '0x64');
+
+      expect(result).toStrictEqual(input);
+    });
+  });
+
+  describe('setPermittedEthChainIds', () => {
+    it('returns a CAIP-25 caveat value with all non-matching EIP-155 scopes removed', () => {
```

Comment:
```suggestion
    it('returns a CAIP-25 caveat value with scopes not listed in chainIds array removed', () => {
```

Reply:
nit!

Reply:
it doesn't remove non-EIP-155 scopes though.

```suggestion
    it('returns a CAIP-25 caveat value with EIP-155 scopes missing from the chainIds array removed', () => {
```

How about this?

Reply:
yep that works!